apps/rush/src/utilities/Git.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import PublishUtilities from './PublishUtilities';
+
+export class Git {
+
+  constructor(private _targetBranch: string) {
+  }
+
+  public checkout(branchName: string, createBranch?: boolean): void {
+    const params: string = `checkout ${createBranch ? '-b ' : ''}${branchName}`;
+
+    PublishUtilities.execCommand(!!this._targetBranch, 'git', params.split(' '));
+  }
+
+  public merge(branchName: string): void {
+    PublishUtilities.execCommand(!!this._targetBranch, 'git', `merge ${branchName} --no-edit`.split(' '));
```

Comment:
This is a refactoring of the code. by moving it out of PublishAction. No code improvement is included in this PR for clearer history.

---
In reply to: [125351522](https://github.com/Microsoft/web-build-tools/pull/259#discussion_r125351522) [](ancestors = 125351522)